### PR TITLE
Branch update dg

### PR DESCRIPTION
**Pull request by kennethk-1201**
Main fix - Update developer guide with my section.

Other Fixes:
1. Update CI banner in README
2. Update some changes to UserGuide to fix error on tp progress

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -110,23 +110,23 @@ Here are the other classes in `Logic` (omitted from the class diagram above) tha
 <img src="images/ParserClasses.png" width="600"/>
 
 How the parsing works:
-* When called upon to parse a user command, the `AddressBookParser` class creates an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `AddressBookParser` returns back as a `Command` object.
+* When called upon to parse a user command, the `AddressBookParser` class creates an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `SudoHRParser` returns back as a `Command` object.
```
Comment (4ndrelim) [inline docs/DeveloperGuide.md:113]: I think 'SudoHRParser' is to replace 'AddressBookParser' in the first line?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -135,18 +135,18 @@ The `Model` component,
...
 
-**API** : [`Storage.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/storage/Storage.java)
+**API** : [`Storage.java`](https://github.com/se-edu/AddressBook-level3/tree/master/src/main/java/seedu/address/storage/Storage.java)
 
 <img src="images/StorageClassDiagram.png" width="550" />
 
 The `Storage` component,
-* can save both address book data and user preference data in json format, and read them back into corresponding objects.
+* can save both system data and user preference data in json format, and read them back into corresponding objects.
 * inherits from both `AddressBookStorage` and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
```
Comment (4ndrelim) [inline docs/DeveloperGuide.md:144]: SudoHRStorage perhaps?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -158,37 +158,37 @@ This section describes some noteworthy details on how certain features are imple
 
 #### Proposed Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `SudoHR` with an undo/redo history, stored internally as an `AddressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
```
Comment (4ndrelim) [inline docs/DeveloperGuide.md:161]: Unsure if 'AddressBookStateList' is intended or if we should change to smth similar to SudoHR

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -158,37 +158,37 @@ This section describes some noteworthy details on how certain features are imple
...
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial system state, and the `currentStatePointer` pointing to that single system state.
 
 ![UndoRedoState0](images/UndoRedoState0.png)
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `delete 5` command to delete the 5th employee in the system. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the system after the `delete 5` command executes to be saved in the `AddressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted system state.
```
Comment (4ndrelim) [inline docs/DeveloperGuide.md:175]: SudoHR for 'AddressBookStateList'?

**Review by 4ndrelim — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (kennethk-1201) [reply docs/DeveloperGuide.md:144]: I think imma just put SudoHR data

**Review by 4ndrelim — APPROVED** [review]
LGTM.

**Pull request merged**